src/main/java/com/redhat/devtools/lsp4ij/actions/LSPShowReformatDialogAction.java

```diff
@@ -0,0 +1,84 @@
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LSPFileSupport;
+import com.redhat.devtools.lsp4ij.LanguageServersRegistry;
+import com.redhat.devtools.lsp4ij.operations.formatting.LSPFormattingSupport;
+import com.redhat.devtools.lsp4ij.operations.formatting.LSPRangeFormattingSupport;
+
+/**
+ * Class overriding the default action handling the Reformat dialog event (CTRL+ALT+SHIFT+L by default)
+ * Fallback to the default action if the language is already supported or not supported by any language server
+ */
+public class LSPShowReformatDialogAction extends ShowReformatFileDialog implements DumbAware {
+    private String HELP_ID = "editing.codeReformatting";
+    private final Logger LOG = Logger.getInstance(LSPShowReformatDialogAction.class);
+
+    @Override
+    public void actionPerformed(AnActionEvent e) {
+        Editor editor = e.getData(CommonDataKeys.EDITOR);
+        Project project = e.getData(CommonDataKeys.PROJECT);
+
+        if (editor != null && project != null) {
+            PsiFile psiFile = PsiDocumentManager.getInstance(project).getPsiFile(editor.getDocument());
+            VirtualFile virFile = FileDocumentManager.getInstance().getFile(editor.getDocument());
+            boolean alreadySupported = !LanguageFormatting.INSTANCE.allForLanguage(psiFile.getLanguage()).isEmpty();
+            if (!alreadySupported && LanguageServersRegistry.getInstance().isFileSupported(psiFile)) {
+                boolean hasSelection = editor.getSelectionModel().hasSelection();
+                LayoutCodeDialog dialog = new LayoutCodeDialog(project, psiFile, hasSelection, HELP_ID);
+                dialog.show();
+                if (dialog.isOK()) {
+                    LayoutCodeOptions options = dialog.getRunOptions();
+                    if (options.getTextRangeType() == TextRangeType.SELECTED_TEXT) {
+                        LSPRangeFormattingSupport rangeFormattingSupport = LSPFileSupport.getSupport(psiFile).getRangeFormattingSupport();
+                        rangeFormattingSupport.format(editor);
+                    } else {
+                        LSPFormattingSupport formattingSupport = LSPFileSupport.getSupport(psiFile).getFormattingSupport();
+                        formattingSupport.format(editor);
+                    }
+                } else {
+                    // if user chose cancel , the dialog in super.actionPerformed(e) will show again
+                    // super.actionPerformed(e);
+                }
+            } else {
+                super.actionPerformed(e);
+            }
+        } else {
+            super.actionPerformed(e);
+        }

```

Comment:
collapse these branches by returning early after the calls to format(editor), and fall back to             super.actionPerformed(e) once at the end;